### PR TITLE
Update linters

### PR DESCRIPTION
**Pull request by ssbarnea**

### CODE REVIEW

**Review by ajinkyau — APPROVED** [review]

File: src/ansiblelint/utils.py [hunk 1]
```diff
@@ -702,7 +702,11 @@ class Task(dict[str, Any]):
     @property
     def name(self) -> str | None:
         """Return the name of the task."""
-        return self.raw_task.get("name", None)
+        name = self.raw_task.get("name", None)
+        if name is not None and not isinstance(name, str):
+            msg = "Task name can only be a string."
+            raise RuntimeError(msg)
```
Comment (cognifloyd) [inline src/ansiblelint/utils.py:708]: Why did you add this / what uncovered this issue? Did someone run into an issue?

Comment (ssbarnea) [reply src/ansiblelint/utils.py:708]: @cognifloyd Issue was reported by mypy, and it makes sense as someone might put some junk data into name. I did not reproduce it, but what mypy reported made sense.
